reference.md

```diff
@@ -0,0 +1,1393 @@
+    Skipping pull part1 (already ran)
+    Skipping pull part2 (already ran)
+    Skipping build part1 (already ran)
+    Skipping build part2 (already ran)
+    Skipping stage part1 (already ran)
+    Skipping stage part2 (already ran)
+    Skipping prime part1 (already ran)
+    Skipping prime part2 (already ran)
+    Snapping 'my-snap' |
+    Snapped my-snap_0_amd64.snap
+
+Add a part and snap again:
+
+    $ snapcraft snap
+    Preparing to pull part3
+    Pulling part3
+    Skipping pull part2 (already ran)
+    Skipping pull part1 (already ran)
+    Preparing to build part3
+    Building part3
+    Skipping build part2 (already ran)
+    Skipping build part1 (already ran)
+    Staging part3
+    Skipping stage part2 (already ran)
+    Skipping stage part1 (already ran)
+    Priming part3
+    Skipping prime part2 (already ran)
+    Skipping prime part1 (already ran)
+    Snapping 'my-snap' /
+    Snapped my-snap_0_amd64.snap
+
+Snap a directory:
+
+     $ snapcraft snap prime
+     Snapping 'my-snap' |
+     Snapped my-snap_0_amd64.snap
+
+Snap specifying the output:
+
+    $ snapcraft snap prime --output /tmp/test.snap
+    Snapping 'my-snap' /
+    Snapped /tmp/test.snap
+
+### cleanbuild
+
+    $ snapcraft cleanbuild
+    Creating snapcraft-insupportably-unrelative-annie
+    Starting snapcraft-insupportably-unrelative-annie
+    Setting up container with project assets
+    ./
+    ./snapcraft.yaml
+    Waiting for a network connection...
+    Network connection established
+    Hit:1 http://archive.ubuntu.com/ubuntu xenial InRelease
+    [...]
+    Reading package lists... Done
+    Building dependency tree
+    Reading state information... Done
+    The following additional packages will be installed:
+    [...]
+    Need to get 14.2 MB of archives.
+    After this operation, 64.6 MB of additional disk space will be used.
+    Get:1 http://archive.ubuntu.com/ubuntu xenial/main amd64 sgml-base all 1.26+nmu4ubuntu1 [12.5 kB]
+    [...]
+    Processing triggers for sgml-base (1.26+nmu4ubuntu1) ...
+    Downloading parts list|
+    Preparing to pull part3
+    Pulling part3
+    Preparing to pull part2
+    Pulling part2
+    Preparing to pull part1
+    Pulling part1
+    Preparing to build part3
+    Building part3
+    Preparing to build part2
+    Building part2
+    Preparing to build part1
+    Building part1
+    Staging part3
+    Staging part2
+    Staging part1
+    Priming part3
+    Priming part2
+    Priming part1
+    Snapping 'my-snap' |
+    Snapped my-snap_0_amd64.snap
+    Retrieved my-snap_0_amd64.snap
+
+### Full lifecycle in a real project
+
+    $ snapcraft/tour/00-SNAPCRAFT/01-easy-start$ snapcraft
+    Preparing to pull gnu-hello
+    Pulling gnu-hello
+    Downloading 'hello-2.10.tar.gz'[=====================================================] 100%
+    Preparing to build gnu-hello
+    Building gnu-hello
+    ./configure --prefix=
+    checking for a BSD-compatible install... /usr/bin/install -c
+    [...]
```

Comment:
This is too verbose, too. We need to default quiet options in the commands, and add verbosity if -d (or maybe a new -v) is passed.